### PR TITLE
[in_app-purchase] Updated the ReadMe to remove the deprecated `InAppPurchaseAndroidPlatformAddition.enablePendingPurchases ` method

### PR DESCRIPTION
**Pull request by dw2kim**
This will resolve flutter/flutter#93837 & this work is just the follow-up from #4527 PR.

We need to update the documentation in the README.md of the in_app_purchase package to remove the instruction of initialization with the deprecated `InAppPurchaseAndroidPlatformAddition.enablePendingPurchases` method.

Fixes https://github.com/flutter/flutter/issues/94977

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

**Comment by dw2kim** [comment]
@mvanbeusekom

**Review by mvanbeusekom — CHANGES_REQUESTED** [review]
Looks good, however you should also update the CHANGELOG.md mentioning the change to the README. Something like:

```md
## 2.0.1

* Removes the instructions on initializing the plugin since this functionality is deprecated.
```

**Review by mvanbeusekom — APPROVED** [review]
LGTM!

**Comment by dw2kim** [comment]
> LGTM, however there seems to be an issue with the CLA. It looks like you pushed commit [8cf9680](https://github.com/flutter/plugins/commit/8cf96801a7aab32d43646205b06c1049f4efe263) with a different email address.

Oh that's why haha. Thank you! will do

File: packages/in_app_purchase/in_app_purchase/README.md [hunk 1]
```diff
@@ -65,29 +64,6 @@ This section has examples of code for the following tasks:
...
-  // An error will occur on Android if you access the plugin `instance`
-  // without this call.
-  if (defaultTargetPlatform == TargetPlatform.android) {
-    InAppPurchaseAndroidPlatformAddition.enablePendingPurchases();
-  }
-  runApp(MyApp());
-}
-```
-
-**Note:** It is not necessary to depend on `com.android.billingclient:billing` in your own app's `android/app/build.gradle` file. If you choose to do so know that conflicts might occur.
```
Comment (stuartmorgan) [inline packages/in_app_purchase/in_app_purchase/README.md:89]: We shouldn't remove this note; it's unrelated to the section above, and it was added based on client feedback.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/in_app_purchase/in_app_purchase/README.md (hunk 1 above)
Comment (dw2kim) [reply packages/in_app_purchase/in_app_purchase/README.md:89]: thank you for the feedback!

**Review by stuartmorgan — APPROVED** [review]
LGTM, thanks!

**Pull request merged**